nbstripout.py

```diff
@@ -143,15 +143,43 @@ def _cells(nb):
             yield cell
 
 
-def strip_output(nb, keep_output, keep_count):
+def pop_recursive(d, key, default=None):
+    """
+    >>> d = {'a': {'b': 1, 'c': 2}}
+    >>> pop_recursive(d, 'a.c')
+    2
+    >>> d
+    {'a': {'b': 1}}
+    """
+    nested = key.split('.')
+    current = d
+    for k in nested[:-1]:
+        current = current.get(k, {})
+    return current.pop(nested[-1], default)
+
+
+def strip_output(nb, keep_output, keep_count, extra_keys=''):
     """
     Strip the outputs, execution count/prompt number and miscellaneous
     metadata from a notebook object, unless specified to keep either the outputs
     or counts.
     """
 
+    # extra_keys could be 'metadata.foo cell.metadata.bar'
```

Comment:
Please document `extra_keys` in the docstring

Reply:
done